src/plugins/trackCodeFlow/varTracking.ts

```diff
@@ -352,13 +352,19 @@ export function createVarLinter(
     }
 
     function finalize(locals: Map<string, VarInfo>) {
-        locals.forEach(local => {
-            if (!local.isUsed && !local.restriction) {
+        // check for unused variables (both locals and function parameters)
+        const allVariables = [...locals.values(), ...args.values()];
```

Comment:
Just thinking about performance a bit here since this whole section can get a bit hot. This code will loop over every local and arg (internally in the js engine) to build the `allVariables` array, and then loop over them _again_ just a moment later. 

Since I've requested that args have a separate rule, could you do the detection in separate loops? Something like:

```
for(const variable of allVariables) { ... }

for(const arg of args){ ... }
```

Reply:
Sure, fixed.